python/src/trezorlib/ui.py

```diff
@@ -172,6 +172,52 @@ def get_passphrase(self, available_on_device: bool) -> Union[str, object]:
                 raise Cancelled from None
 
 
+class ScriptUI:
+    """Interface to be used by scripts, not directly by user.
+
+    Communicates with a client application using print() and input().
+
+    Lot of `ClickUI` logic is outsourced to the client application, which
+    is responsible for supplying the PIN and passphrase.
+
+    Reference client implementation can be found under `cli/test_script_client.py`.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def button_request(self, br: messages.ButtonRequest) -> None:
+        name = br.code.name if br.code else None
```

Comment:
are there cases when `br.code` is None?

if not, `code` should be `br.code.name`, and `name` should be `br.name` -- this doesn't exist yet, so maybe we can leave it out for now

Reply:
I could not see any case where it is `None`, but it is marked `Optional` in protobuf.

`br.name` is indeed not defined, so I will delete the `name=xxx` part. We could probably leave it in the client, it is `Optional` there

Reply:
Name part deleted in e5670a4